File: back-end/src/main/java/pl/straburzynski/campfiresongs/session/model/Session.java
package pl.straburzynski.campfiresongs.session.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.Type;
import pl.straburzynski.campfiresongs.song.model.Song;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import java.util.UUID;

@Entity
@Table(name = "SESSION")
@Builder
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Session {

    @Id
    @Column(name = "ID", nullable = false)
    @Type(type = "uuid-char")
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "SONG_ID")
    private Song song;

    @Column(name = "NAME", nullable = false)
    private String name;

    @Column(name = "PASSWORD")
    private String password;

    @Column(name = "TEMPORARY")
    private boolean temporary = false;

    @Column(name = "SONG_DATA", length = Integer.MAX_VALUE)
    private String songData;

}

File: back-end-kotlin/src/main/kotlin/pl/straburzynski/campfiresongs/session/model/Session.kt
package pl.straburzynski.campfiresongs.session.model

import jakarta.persistence.CascadeType.ALL
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import org.hibernate.annotations.UuidGenerator
import pl.straburzynski.campfiresongs.song.model.Song
import java.util.UUID


@Entity
@Table(name = "SESSIONS")
data class Session(
    @Id
    @Column(name = "ID", nullable = false)
    @UuidGenerator
    var id: UUID? = null,

    @ManyToOne(fetch = FetchType.LAZY, cascade = [ALL])
    @JoinColumn(name = "SONGS_ID")
    var song: Song? = null,

    @Column(name = "NAME")
    val name: String,

    @Column(name = "PASSWORD")
    val password: String,

    @Column(name = "TEMPORARY")
    var temporary: Boolean = false,

    @Column(name = "SONG_DATA", length = Int.MAX_VALUE)
    var songData: String? = null
)
